src/ConfigManager.h

```diff
@@ -106,7 +106,7 @@ class ConfigManager {
     static void SetEventWindow(G4double value)              { Instance()->eventWindow_ = value; }
     static void SetSNTimingOn(G4bool value)                 { Instance()->snTimingOn_ = value; }
     static void SetTh2Name(G4String value)                  { Instance()->th2Name_ = value; }            
-    static void SetUseHDDetectorConfiguration(G4bool value) { Instance()->useHDDetectorConfiguration_ = value; }
+    static void SetDetectorConfiguration(G4String value)    { Instance()->detectorConfiguration_ = value; }
```

Comment:
For non-built-in data types, pass by reference, never by value (unless you have a really good reason to need by-value passing).  Here, use `const G4String& value`.

Reply:
Will revise this line. Thank you Mike!